security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/PhaseTapChangerRegulationAction.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.security.action;
+
+import com.powsybl.iidm.network.PhaseTapChanger;
+import com.powsybl.iidm.network.ThreeWindingsTransformer;
+
+import java.util.Optional;
+
+/**
+ * An action modifying the regulation of a phase-shifting transformer
+ *
+ * @author Etienne Lesot <etienne.lesot@rte-france.com>
+ * @author Anne Tilloy <anne.tilloy@rte-france.com>
+ */
+public class PhaseTapChangerRegulationAction extends AbstractTapChangerRegulationAction {
+
+    public static final String NAME = "PHASE_TAP_CHANGER_REGULATION";
+    private final PhaseTapChanger.RegulationMode regulationMode;
+
+    public PhaseTapChangerRegulationAction(String id, String transformerId, ThreeWindingsTransformer.Side side, boolean regulating, PhaseTapChanger.RegulationMode regulationMode) {
+        super(id, transformerId, side, regulating);
+        this.regulationMode = regulationMode;
+        if (!regulating && this.regulationMode != null) {
+            throw new IllegalArgumentException("PhaseTapChangerRegulationAction can not have a regulation mode " +
+                    "if it is not regulating");
+        }
+    }
+
+    @Override
+    public String getType() {
+        return NAME;
+    }
+
+    public Optional<PhaseTapChanger.RegulationMode> getRegulationMode() {
+        return Optional.ofNullable(regulationMode);
+    }
+
+    public static PhaseTapChangerRegulationAction activateRegulation(String id, String transformerId, PhaseTapChanger.RegulationMode regulationMode) {
```

Comment:
I think that you need:
- 2 `activateRegulation` methods for two and three windings transformer when you only change the boolean and you keep the regulation mode to the initial value.
- Maybe 2 methods that change the regulation mode. When you do that, you must set the regulating boolean to true, change the mode and change the target value.